massa-network-worker/src/handshake_worker.rs

```diff
@@ -31,12 +31,67 @@ use massa_network_exports::{
 use massa_signature::KeyPair;
 use massa_time::MassaTime;
 use rand::{rngs::StdRng, RngCore, SeedableRng};
-use tokio::{task::JoinHandle, time::timeout};
-use tracing::debug;
+use std::collections::HashMap;
+use std::thread::{self, JoinHandle};
+use tokio::runtime::Handle;
+use tokio::time::timeout;
 
 /// Type alias for more readability
 pub type HandshakeReturnType = Result<(NodeId, ReadBinder, WriteBinder), NetworkError>;
 
+/// Start a thread, that will be responsible for running handshake workers,
+/// and sending the results back to the network worker.
+pub fn start_handshake_manager(
```

Comment:
This is an example of a new structure, used to manage a separate concern, in this case running handshakes. It is started by the network worker, but runs separately from it, and communications results of handshakes back. It also owns a handle to a tokio runtime to run the actual handshake. 

Note: it would be best to separate the cpu intensive signature and serialization, from the other networking things happening in an handshake, and do the cpu intensive things on a thread(perhaps a shared "signer" and "serializer").